utils/parse.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const strip = ({ id, username }) =>
+	id ? { id } : { username: username.toLowerCase() };
+
+const toUserObject = s =>
+	s.user || (/^\d+$/.test(s) ? { id: +s } : { username: s.replace('@', '') });
+
+const isTextMention = m => m.type === 'text_mention';
+
+const spliceOut = (s, { offset, length }) =>
+	s.slice(0, offset) + s.slice(offset + length);
+
+const parse = message => {
+	// eslint-disable-next-line no-empty-character-class
+	const regex = /^\/\w+(?:@\w+)?((?:@\w+|\d+|\s+)*)(.*)$/s;
```

Comment:
What on earth is this regex for?

Reply:
It splits message text into 3 parts -- command, ids and mentions, and reason.